### DISCUSSION TITLE
Stream a PNG files using SIP Sorcery to generate a video output on client

### DISCUSSION THREAD

**Discussion by jadavparesh06** [sipsorcery-org/sipsorcery#628] [Q&A]
I was following SIP Sorery library GetStarted example where it is streaming I420 Test Pattern to client, My questions is If I have a png file instead of a I420, how can we stream it.

Basically, I will be getting frame by frame video data at my end as png files and I want to stream them to client so that it recreates a video using those files.

**Comment by sipsorcery** [comment]
The [VideoTestPatternSource](https://github.com/sipsorcery-org/sipsorcery/blob/fc90501587bd49b661a833dc19f034d13614c5ad/src/app/Media/Sources/VideoTestPatternSource.cs#L225) is an example of generating a video stream from static images.

You can convert a png image to a I420 using `FFmpeg` and `-pix_fmt yuvj420p`.

**Reply by jadavparesh06** [reply]
Any other alternative than FFmpeg for this conversion?

Also I have observed one weird behavior when I use a bmp file as BGR format and try to stream it, the image content shows flipped vertically.
